app/main.ts

```diff
@@ -3021,6 +3023,32 @@ ipc.handle('show-save-dialog', async (_event, { defaultPath }) => {
     return { canceled: true };
   }
 
+  if (
+    process.platform === 'linux' &&
+    defaultPath &&
+    defaultPath.charAt(0) !== '/'
+  ) {
+    // On Linux the defaultPath should be an absolute path, otherwise the save dialog will have an empty filename on KDE/Plasma
+    let downloadsPath = '';
+    try {
+      downloadsPath = (
+        await promisify(exec)('xdg-user-dir DOWNLOAD')
+      ).stdout.trim();
+      getLogger().info(
+        'show-save-dialog: saving to user downloads directory: ' + downloadsPath
+      );
+    } catch (e) {
+      // If we cannot get Downloads path, fall back to the user's home directory
+      downloadsPath = process.env['HOME'] || '';
+      getLogger().info(
+        'show-save-dialog: saving to user home directory: ' + downloadsPath
+      );
+    }
```

Comment:
You want to use `app.getPath('downloads')` and `app.getPath('home')` here from the `electron` module.

Reply:
Huh, that's convenient. Done.